sympy/tensor/array/arrayop.py

```diff
@@ -208,3 +208,38 @@ def derive_by_array(expr, dx):
             return ImmutableDenseNDimArray([expr.diff(i) for i in dx], dx.shape)
         else:
             return diff(expr, dx)
+
+
+def permutedims(expr, perm):
+    """
+    Permutes the indices of an array.
+
+    Parameter specifies the permutation of the indices.
+    """
+    if not isinstance(expr, NDimArray):
+        raise TypeError("expression has to be an N-dim array")
+
+    perm = tuple(perm)
```

Comment:
This change won't lead to any complications for end-user - he can still pass tuple or list to the `perm` argument.

Two main reasons for this change:
- Do not reinvent the wheel (inverse & check for validity of permutation).
- Allow to use not only tuples, but also Permutation objects. And anything else that `Permutation` constructor accepts.